### ISSUE TITLE
sk Toolkit: Log Message Format 

### ISSUE THREAD

**Issue by stormecharette** [maine-cyber/mcsc-unisim#64]
The *sktk* uses its own logger to report its errors, warnings, and notes. The logger automatically tags log messages with either `<ERROR>`, `<WARNING>`, `<STATUS>`, `<DEBUG>`, `<NOTE>`, as well as the current time so that certain log message types can easily be caught with regular expressions or other parsing programs.  

However, the message part of the string has yet to be standardized for the *sktk*. It is hard to distinguish *sk* errors from other errors in a program, and some error messages either miss crucial details, or provide unnecessary ones. 

If would be nice if we could decide on a standard message header for the *sktk*. It needs to be **really concise**.

**Comment by stormecharette** [comment]
While working on some bug fixes I encountered one possible format for *sktk* log messages.  

The first thing in the message is the name of the **module** producing the message in the format `-sk [MODULE]-`, then the first word of the message should be the name of the **object or function** producing the message, followed by the message itself.    

For example:   

`mLOG :: [08:12:55] <ERROR> :::: -sk Config- ConfigReader failed and should feel ashamed.`  

I find this pretty readable, and the `-sk *-` pattern allows for quick searches for messages from specific modules. Also the guarantee that the object or function's symbol will appear in the log allows for better searches too.   

@ksabochick, what do you think?

**Comment by ksabochick** [comment]
I agree with including the specific module producing the message - clears up confusion and makes it easy to trace the error. Would it make sense to also include the reaction to the bug? For example, if the program terminated because of a failure, it would be included, maybe as "STATUS" or something.

`mLOG :: [11:54:47] <ERROR> :::: -sk Network- -STATUS: Terminated- LocalDatagram failed because it didn't study`

**Comment by stormecharette** [comment]
It's been decided... we will be using the log format:  

`"-sk [MODULE]- [CLASS OR FUNCTION] [MSG]"`  

for shipped *sktk* log messages from now on. Now we have to rewrite every log in the toolkit.  

- [ ] **Rewrite existing logs with new format.**
